test/kv_integration_test.go

```diff
@@ -684,4 +684,37 @@ func (kvSuite *KvTestSuite) TestRequestSystemAttrs() {
 
 	err = it.Err()
 	kvSuite.NoError(err)
-}
\ No newline at end of file
+}
+func (kvSuite *KvTestSuite) TestNonExistingColumns() {
+	table := fmt.Sprintf("TestNonExistingColumns%d", time.Now().UnixNano())
+
+	frame := kvSuite.generateRandomSampleFrame(5, "idx", []string{"n1", "n2"})
+	wreq := &frames.WriteRequest{
+		Backend: kvSuite.backendName,
+		Table:   table,
+	}
+
+	appender, err := kvSuite.client.Write(wreq)
+	if err != nil {
+		kvSuite.T().Fatal(err)
+	}
+
+	if err := appender.Add(frame); err != nil {
+		kvSuite.T().Fatal(err)
+	}
+
+	if err := appender.WaitForComplete(3 * time.Second); err != nil {
+		kvSuite.T().Fatal(err)
+	}
+
+	kvSuite.T().Log("read")
+	rreq := &pb.ReadRequest{
+		Backend: kvSuite.backendName,
+		Table:   table,
+		Columns: []string{"n1", "kk"},
+	}
+
+	it, _ := kvSuite.client.Read(rreq)
+	it.Next()
+	kvSuite.NotEmptyf(it.Err(), "error was expected when reading a non existing column")
```

Comment:
better use `kvSuite.Error(it.Err(),  "error was expected when reading a non existing column" )`